main.go

```diff
@@ -136,3 +141,13 @@ func getLoggerWithLogLevel(logLevel string, logFilePath string) (logr.Logger, er
 	ctrlLogger := logger.New(logLevel, logFilePath)
 	return zapr.NewLogger(ctrlLogger), nil
 }
+
+// validate controller flags
+func validControllerFlags(ctrlConfig config.ControllerConfig) bool {
```

Comment:
Probably better to move it to `controller_config.go` and return an error if the validation fails and the err check above should catch it.